src/Microsoft.SqlTools.ServiceLayer/Rename/RenameService.cs

```diff
@@ -0,0 +1,138 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+using System;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient;
+using Microsoft.SqlServer.Management.Common;
+using Microsoft.SqlTools.Hosting.Protocol;
+using Microsoft.SqlTools.ServiceLayer.Connection;
+using Microsoft.SqlTools.ServiceLayer.Rename.Requests;
+using Microsoft.SqlTools.Utility;
+
+namespace Microsoft.SqlTools.ServiceLayer.Rename
+{
+    /// <summary>
+    /// Main class for Rename Service functionality
+    /// </summary>
+    public class RenameService : IDisposable
+    {
+        private bool disposed = false;
+
+        private static Lazy<RenameService> renameServiceInstance = new Lazy<RenameService>(() => new RenameService());
+        public static RenameService Instance => renameServiceInstance.Value;
+
+        public static ConnectionService connectionService;
+
+        private IProtocolEndpoint serviceHost;
+
+        public RenameService() { }
+        /// <summary>
+        /// Internal for testing purposes only
+        /// </summary>
+        internal static ConnectionService ConnectionServiceInstance
+        {
+            get
+            {
+                if (connectionService == null)
+                {
+                    connectionService = ConnectionService.Instance;
+                }
+                return connectionService;
+            }
+            set
+            {
+                connectionService = value;
+            }
+        }
+
+        public void InitializeService(IProtocolEndpoint serviceHost)
+        {
+            this.serviceHost = serviceHost;
+            this.serviceHost.SetRequestHandler(ProcessRenameEditRequest.Type, HandleProcessRenameEditRequest);
+
+        }
+        internal Task HandleRequest<T>(RequestContext<T> requestContext, Func<Task> action)
+        {
+            // The request handling will take some time to return, we need to use a separate task to run the request handler so that it won't block the main thread.
```

Comment:
looks like you copied the code from table designer :)

Reply:
That's where i got the "inspiration" of how the projects works 😉